mgmt/FileManager.cc

```diff
@@ -602,9 +600,9 @@ FileManager::WalkSnaps(ExpandingArray *snapList)
 {
   MFresult r;
 
-  // The original code reset this->managedDir from proxy.config.snapshot_dir at this point. There doesn't appear to be
-  // any need for that, since managedDir is always set in the constructor and should not be changed.
-  ink_release_assert(this->managedDir != NULL);
+  // Make sure managedDir is the latest from proxy.config.snapshot_dir.
+  ats_scoped_str snapshotDir(RecConfigReadSnapshotDir());
+  this->managedDir = snapshotDir;
```

Comment:
You'll need to reset managedDir to NULL after the call to ``WalkFiles`` so you don't leave it as a dangling pointer.